xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -1552,24 +1552,16 @@ public Date getResourceLastModificationDate(String name)
 
     public byte[] getResourceContentAsBytes(String name) throws IOException
     {
-        InputStream is = null;
+        byte output[] = new byte[0];
         if (getEngineContext() != null) {
-
-            try {
-                is = getResourceAsStream(name);
-            } catch (Exception e) {
+            try (InputStream is = getResourceAsStream(name)) {
```

Comment:
Actually I mixed with the other pull request, there was indeed a catch before so you should put it back, sorry.

Reply:
One difficulty is that readFileToByteArray itself is not supposed to be catched.

Reply:
Well, in order to use the try-with-resources block we will have to put it in the try block right? Maybe we could find the specific exception thrown by the input stream to avoid catching anything for readFileToByteArray?

Reply:
readFileToByteArray does not nee the InputStream so no you don't need to put it in the try.

Reply:
@tmortagne okay, I removed it